### PR TITLE
For #78: Errors dynamo repository implementation.

### PR DESCRIPTION
**Pull request by paulodamaso**
For #78:
- Implemented `DyError` and `DyErrors` classes
- Added todo for creating `errors` table on database

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#81](https://github.com/yegor256/wring/issues/81) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#81](https://github.com/yegor256/wring/issues/81) is assigned to @pinaf[/z](https://www.0crat.com/u/pinaf), here is [why](https://www.0crat.com/footprint/C7FCB1EQN/94e67e4d-0b26-43b1-b33f-42abb76b885c); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/io/wring/dynamo/DyError.java [hunk 1]
```diff
@@ -40,14 +46,64 @@
...
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
 public final class DyError implements Error {
+    /**
+     * The item.
+     */
+    private final transient Item item;
+
+    /**
+     * Ctor.
+     * @param itm Item with pitch
```
Comment (pinaf) [inline src/main/java/io/wring/dynamo/DyError.java]: @paulodamaso pitch?

File: src/main/java/io/wring/dynamo/DyError.java [hunk 2]
```diff
@@ -40,14 +46,64 @@
...
+        );
+    }
+
+    /**
+     * To HTML.
+     * @param text Text in Markdown (simplified)
+     * @return HTML
+     */
+    private static String html(final CharSequence text) {
+        final Pattern ptn = Pattern.compile("\\[([^]]+)]\\(([^)]+)\\)");
```
Comment (pinaf) [inline src/main/java/io/wring/dynamo/DyError.java]: @paulodamaso where does this pattern come from? what does it mean?

File: src/main/java/io/wring/model/Errors.java [hunk 3]
```diff
@@ -56,5 +56,5 @@
      * @param description Description
      * @throws IOException If fails
      */
-    void register(String title, String description);
+    void register(String title, String description) throws IOException;
```
Comment (pinaf) [inline src/main/java/io/wring/model/Errors.java]: @paulodamaso wouldn't it be better to thrown an unchecked exception here instead?

File: src/main/java/io/wring/dynamo/DyError.java (hunk 1 above)
Comment (paulodamaso) [reply src/main/java/io/wring/dynamo/DyError.java]: @pinaf Just copied from `DyEvent`, sorry.

File: src/main/java/io/wring/dynamo/DyError.java (hunk 2 above)
Comment (paulodamaso) [reply src/main/java/io/wring/dynamo/DyError.java]: @pinaf Actually I've just copied it from `DyEvent` but I think that we do not need to use it here, I'll remove it.

**Comment by paulodamaso** [comment]
@pinaf Corrections done, please take a look.

File: src/main/java/io/wring/dynamo/DyErrors.java [hunk 4]
```diff
@@ -70,12 +77,63 @@ public DyErrors(final Region reg, final String user) {
...
+                    .with("title", title)
+                    .with("description", description)
+                    .with("time", System.currentTimeMillis())
+            );
+            Logger.info(
+                this, "Error registered for %s: \"%s\"",
+                this.urn, title
+            );
+        } catch (final IOException err) {
+            throw new RuntimeException(err);
```
Comment (pinaf) [inline src/main/java/io/wring/dynamo/DyErrors.java]: @paulodamaso let's go with the non-raw `IllegalStateException` here.

File: src/main/java/io/wring/dynamo/DyErrors.java [hunk 5]
```diff
@@ -70,12 +77,63 @@ public DyErrors(final Region reg, final String user) {
...
+            )
+            .where("urn", Conditions.equalTo(this.urn))
+            .stream()
+            .map(DyError::new)
+            .map(Error.class::cast)
+            .iterator();
     }
 
     @Override
+    @SuppressWarnings("PMD.AvoidThrowingRawExceptionTypes")
```
Comment (pinaf) [inline src/main/java/io/wring/dynamo/DyErrors.java]: @paulodamaso then you can probably get rid of this.

**Comment by pinaf** [comment]
@paulodamaso thanks. a couple more.

**Comment by paulodamaso** [comment]
@carlosmiranda Updated class, please try to merge again

**Comment by carlosmiranda** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @pinaf[/z](https://www.0crat.com/u/pinaf), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#81](https://github.com/yegor256/wring/issues/81) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda)

**Comment by ypshenychka** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +20 point(s) just awarded to @pinaf[/z](https://www.0crat.com/u/pinaf)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**